server/server.js

```diff
@@ -129,6 +165,83 @@ app.post('/api/sign-in', async (req, res, next) => {
   }
 });
 
+// User can add to shopping cart
+app.post(
+  '/api/orderItems/:orderItemId',
+  authMiddleware,
+  async (req, res, next) => {
+    try {
+      const { orderId, productId, quantity } = req.body;
+      if (!orderId || !productId || !quantity)
+        throw new ClientError(
+          400,
+          'UserID, productID, and quantity are required fields'
+        );
+      const sql = `
+    insert into "orderItems" ("orderId", "productId", "quantity")
+    values ($1, $2, $3)
+    returning *;
+    `;
+      const params = [orderId, productId, quantity];
+      const result = await db.query(sql, params);
+      const [shoppingcart] = result.rows;
+      res.status(201).json(shoppingcart);
+    } catch (error) {
+      next(error);
+    }
+  }
+);
+
+// User can update shopping cart
+app.put(
+  '/api/orderItems/:orderItemId',
+  authMiddleware,
+  async (req, res, next) => {
+    try {
+      const { orderId, productId, quantity } = req.body;
+      if (!quantity) throw new ClientError(400, 'Quantity is required');
+      const sql = `
+    update "orderItems"
+    set "quantity" = $3
+    where "orderId" = $1 and "productId" = $2
+    returning *;
+    `;
+      const params = [orderId, productId, quantity];
+      const result = await db.query(sql, params);
+      const order = result.rows;
+      res.status(201).json(order);
+    } catch (error) {
+      next(error);
+    }
+  }
+);
+
+// User can delete product from shopping cart
+app.delete(
+  '/api/orderItems/:orderItemId',
+  authMiddleware,
+  async (req, res, next) => {
+    try {
+      const orderItemId = Number(req.params.orderItemId);
+      if (!Number.isInteger(orderItemId))
+        throw new ClientError(400, 'OrderItemID must be an integer.');
+      const sql = `
+    delete from "orderItems"
+    where "orderId" = $1 and "productId" = $2
```

Comment:
This one's a bit odd. You haven't been passed an orderId or productId, so you can't use them. However, you do have the orderItemId, so you can use it: `where "orderItemId" = $1` and pass `orderItemId` as the only parameter.